### PR TITLE
fix: Assign agent action changes

### PR DESCRIPTION
**Pull request by tejaswinichile**
Fixes https://github.com/chatwoot/product/issues/653

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested on local.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/services/action_service.rb [hunk 1]
```diff
@@ -47,8 +47,9 @@ def send_email_transcript(emails)
 
   private
 
-  def agent_belongs_to_account?(agent_ids)
-    @account.agents.pluck(:id).include?(agent_ids[0])
+  def agent_belongs_to_inbox?(agent_ids)
+    member_ids = @conversation.inbox.members.pluck(:user_id)
```
Comment (pranavrajs) [inline app/services/action_service.rb:51]: This + admins of the account can be added to the conversation.. I thought that was clear, My mistake for not communicating the permissions properly. The intention was to check whether the agent was assignable or not.

**Comment by fayazara** [comment]
@tejaswinichile @pranavrajs We should show some sort of warning/alert when someone runs a macro which has assign_agent, but the agent is not linked to the inbox.

A error message stating something like `<AGENT_NAME)> in this macro action is not linked to this inbox - hence cannot fulfil this macro`.

**Comment by fayazara** [comment]
> @tejaswinichile @pranavrajs We should show some sort of warning/alert when someone runs a macro which has assign_agent action, but the agent is not linked to the inbox.
> 
> A error message stating something like `<AGENT_NAME)> in this macro action is not linked to this inbox - hence cannot fulfil this macro`.

Right now, it just shows "Macro executed successfully" - with a success response 200.

**Comment by tejaswinichile** [comment]
@fayazara I checked, and we can't add this as an error message.

1. As we run it in the backend sidekiq job.
2. And if we add errors we stop entire the execution of other actions in a macro.

**Review by pranavrajs — APPROVED** [review]
@tejaswinichile Looks fine. Don't we need a migration to migrate the existing values?

**Comment by pranavrajs** [comment]
@fayazara 

> We should show some sort of warning/alert when someone runs a macro that has assign_agent action, but the agent is not linked to the inbox.

Let us make this as an enhancement.

**Comment by pranavrajs** [comment]
@muhsin-k Can you please do a QA change?

**Review by muhsin-k — APPROVED** [review]
Tested in the staging environment.

https://www.loom.com/share/a83170d144dd4e3b8382d1ff71fd8530

**Pull request merged**